keps/sig-node/4817-resource-claim-device-status/README.md

```diff
@@ -0,0 +1,618 @@
+by using the Kubernetes API.
+
+```golang
+// ResourceClaimStatus tracks whether the resource has been allocated and what
+// the result of that was.
+type ResourceClaimStatus struct {
+    ...
+
+    // DeviceStatuses contains the status of each device allocated for this
+    // claim, as reported by the driver. This can include driver-specific
+    // information. Entries are owned by their respective drivers.
+    //
+    // +optional
+    // +listType=map
+    // +listMapKey=driver
+    // +listMapKey=device
+    // +listMapKey=pool
+    DeviceStatuses []AllocatedDeviceStatus `json:"deviceStatuses,omitempty" protobuf:"bytes,4,opt,name=deviceStatuses"`
+}
+
+
+// AllocatedDeviceStatus contains the status of an allocated device, if the
+// driver chooses to report it. This may include driver-specific information.
+type AllocatedDeviceStatus struct {
+    // Driver specifies the name of the DRA driver whose kubelet
+    // plugin should be invoked to process the allocation once the claim is
+    // needed on a node.
+    //
+    // Must be a DNS subdomain and should end with a DNS domain owned by the
+    // vendor of the driver.
+    //
+    // +required
+    Driver string `json:"driver" protobuf:"bytes,2,rep,name=driver"`
+
+    // This name together with the driver name and the device name field
+    // identify which device was allocated (`<driver name>/<pool name>/<device name>`).
+    //
+    // Must not be longer than 253 characters and may contain one or more
+    // DNS sub-domains separated by slashes.
+    //
+    // +required
+    Pool string `json:"pool" protobuf:"bytes,3,rep,name=pool"`
+
+    // Device references one device instance via its name in the driver's
+    // resource pool. It must be a DNS label.
+    //
+    // +required
+    Device string `json:"device" protobuf:"bytes,4,rep,name=device"`
+
+    // Conditions contains the latest observation of the device's state.
+    // If the device has been configured according to the class and claim
+    // config references, the `Ready` condition should be True.
+    //
+    // +optional
+    // +listType=atomic
+    Conditions []metav1.Condition `json:"conditions" protobuf:"bytes,5,rep,name=conditions"`
+
+    // DeviceInfo contains arbitrary driver-specific data.
+    //
+    // +optional
+    // +listType=atomic
+    DeviceInfo []runtime.RawExtension `json:"deviceInfo,omitempty" protobuf:"bytes,6,rep,name=deviceInfo"`
+
+    // NetworkDeviceInfo contains network-related information specific to the device.
+    //
+    // +optional
+    NetworkDeviceInfo NetworkDeviceInfo `json:"networkDeviceInfo,omitempty" protobuf:"bytes,7,rep,name=networkDeviceInfo"`
+}
+
+// NetworkDeviceInfo provides network-related details for the allocated device.
+// This information may be filled by drivers or other components to configure
+// or identify the device within a network context.
+type NetworkDeviceInfo struct {
```

Comment:
This comment seems to mean "does 'network device' mean a whole 'network'?", at least to me. At least to me, I suppose it does not.

'A network device' is a device that connects to 'a network', so a network device could be just a part of ' a network'.

Other cloud infrastructures tend to use 'network interface' for such network device information, such as
- https://cloud.google.com/vpc/docs/create-use-multiple-interfaces#creating_virtual_machine_instances_with_multiple_network_interfaces
- https://docs.aws.amazon.com/cli/latest/reference/ec2/describe-network-interfaces.html

hence, user may get confused if Kubernetes uses the term, 'network', for 'network interface' information because they may expect 'network interface' or 'network device' for such information.

If we call 'NetworkInfo', then it should contain 'network address', not 'network interface address' because it is 'NetworkInfo'.

Reply:
Well..."NetworkInfo" could mean "the network info associated with this device"...since we are already narrowed down the context to a device.